src/server/game/Spells/Spell.cpp

```diff
@@ -2826,7 +2826,7 @@ void Spell::DoSpellEffectHit(Unit* unit, uint8 effIndex, TargetInfo& hitInfo)
 
             if (!hitInfo.HitAura)
             {
-                bool const resetPeriodicTimer = !(_triggeredCastFlags & TRIGGERED_DONT_RESET_PERIODIC_TIMER);
+                bool const resetPeriodicTimer = !(_triggeredCastFlags & TRIGGERED_DONT_RESET_PERIODIC_TIMER) && !hitInfo.AuraSpellInfo->HasAttribute(SPELL_ATTR0_CU_DONT_RESET_PERIODIC_TIMER);
```

Comment:
i tested with debug, all spells tested (with aura effect) hitInfo.AuraSpellInfo is not null at this point
and cases without aura, dont even reach this point...